browser/brave_ads/notification_helper/notification_helper.h

```diff
@@ -1,32 +1,52 @@
-/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #ifndef BRAVE_BROWSER_BRAVE_ADS_NOTIFICATION_HELPER_NOTIFICATION_HELPER_H_
 #define BRAVE_BROWSER_BRAVE_ADS_NOTIFICATION_HELPER_NOTIFICATION_HELPER_H_
 
+#include <memory>
+
+#include "base/memory/weak_ptr.h"
+
+class Profile;
+
+namespace base {
+template <typename Type>
+struct DefaultSingletonTraits;
+}  // namespace base
+
 namespace brave_ads {
 
-class NotificationHelper {
- public:
-  NotificationHelper(const NotificationHelper&) = delete;
-  NotificationHelper& operator=(const NotificationHelper&) = delete;
-  virtual ~NotificationHelper();
+class NotificationHelperImpl;
 
+class NotificationHelper final {
+ public:
   static NotificationHelper* GetInstance();
 
-  virtual bool CanShowNativeNotifications();
-  virtual bool CanShowNativeNotificationsWhileBrowserIsBackgrounded() const;
+  void InitForProfile(Profile* profile);
 
-  virtual bool ShowOnboardingNotification();
+  bool CanShowNativeNotifications();
+  bool CanShowNativeNotificationsWhileBrowserIsBackgrounded() const;
 
- protected:
-  friend class NotificationHelperHolder;
+  bool ShowOnboardingNotification();
+
+  bool SystemNotificationsSupported() const;
```

Comment:
Could this not be part of `CanShowNativeNotifications`? As CanShow should only return true if native/system notifications are supported

Reply:
Renamed `CanShowNativeNotifications` to `CanShowNotifications` as it is true for both system and chrome message center notifications.
Renamed `CanShowNativeNotificationsWhileBrowserIsBackgrounded` to `CanShowSystemNotificationsWhileBrowserIsBackgrounded`.